storage/innobase/handler/ha_innodb.cc

```diff
@@ -7066,6 +7066,7 @@ ha_innobase::open(
 			  srv_pass_corrupt_table <= 1)) {
 
 		free_share(m_share);
+		dict_table_close(ib_table, FALSE, FALSE);
```

Comment:
I think this is good idea to call dict_table_close() on error paths. We have increased the ref_count in open_table() and for intrinsic tables, we did a acquire() which increases ref_count.

dict_table_close() can handle intrinsic tables too. But we have to be careful in passing NULL to dict_table_close(). Some paths set ib_table to NULL. and following dict_table_close(ib_table) can cause crash.

Just analyze and ensure dic_table_close(NULL) doesn't happen.

Reply:
Yes, I have double checked that `dict_table_close` is provided a not null `ib_table`.

Reply:
Great!